services/src/main/scala/cromwell/services/metadata/impl/MetadataDatabaseAccess.scala

```diff
@@ -125,6 +125,7 @@ trait MetadataDatabaseAccess {
       case MetadataQuery(_, None, None, Some(includeKeys), Some(excludeKeys), _) => Future.failed(
         new IllegalArgumentException(
           s"Include/Exclude keys may not be mixed: include = $includeKeys, exclude = $excludeKeys"))
+      case _ => Future.failed(new IllegalStateException(s"Invalid MetadataQuery: $query"))
```

Comment:
I chose `InvalidStateException` because this is covering a case which the developer(s) seemed to believe wasn't possible (and clearly has never happened before since it hasn't come up).

The 2.12 linter is more robust to detecting non-exhaustive matching